nuxeo-compound-documents/src/test/java/org/nuxeo/compound/documents/TestCompoundDocument.java

```diff
@@ -16,65 +16,95 @@
 @Deploy("org.nuxeo.ecm.platform.audio.core:OSGI-INF/core-types-contrib.xml")
-@Deploy("org.nuxeo.ecm.platform.tag:OSGI-INF/faceted-tag-service-core-types.xml")
 @Deploy("org.nuxeo.ecm.platform.picture.core:OSGI-INF/picture-schemas-contrib.xml")
-@Deploy("org.nuxeo.ecm.platform.types:OSGI-INF/subtypes-enricher-contrib.xml")
 @Deploy("org.nuxeo.compound.documents")
-public class TestCompoundDocument extends AbstractJsonWriterTest.Local<DocumentModelJsonWriter, DocumentModel> {
+public class TestCompoundDocument extends BaseTest {
 
     @Inject
     protected CoreSession session;
-    
-    public TestCompoundDocument() {
-        super(DocumentModelJsonWriter.class, DocumentModel.class);
+
+    @Inject
+    protected TransactionalFeature txFeature;
+
+    @Inject
+    protected ServletContainerFeature servletContainerFeature;
+
+    protected HttpClientTestRule httpClientRule;
+
+    protected ObjectMapper mapper = new ObjectMapper();
+
+    @Test
+    public void testCreatecompoundDocument() throws IOException {
+        String data = "{\"entity-type\": \"document\",\"type\": \"CompoundDocument\", \"name\": \"myCompoundDocument\", \"properties\": {\"dc:title\": \"My title\", \"dc:description\": \" \"}}";
+        try (CloseableClientResponse response = getResponse(RequestType.POST, "path/", data)) {
+            assertEquals(SC_CREATED, response.getStatus());
+        }
+        var headers = Map.of(MarshallingConstants.EMBED_ENRICHERS + ".document", SubtypesJsonEnricher.NAME);
+        try (CloseableClientResponse response = getResponse(RequestType.GET, "path/myCompoundDocument", headers)) {
+            assertCoumpoundResponse(response, "CompoundDocument");
+        }
     }
 
     @Test
-    public void testCompoundDocument() throws IOException {
-        DocumentModel doc = session.createDocument(session.createDocumentModel("/", "test", "CompoundDocument"));
-        assertTrue(doc.getFacets().contains("CompoundDocument"));
-        RenderingContext ctx = RenderingContext.CtxBuilder.enrichDoc("subtypes").get();
-        JsonAssert json = jsonAssert(doc, ctx);
-        json = json.has("contextParameters").isObject();
-        json.properties(1);
-        json = json.has("subtypes").isArray();
-        json.childrenContains("type", "File", "Picture", "Video", "Audio", "CompoundDocumentFolder");
+    public void testCreateCompoundDocumentFolder() throws IOException {
+        session.createDocument(session.createDocumentModel("/", "myCompoundDocument", "CompoundDocument"));
+        txFeature.nextTransaction();
+        String data = "{\"entity-type\": \"document\",\"type\": \"CompoundDocumentFolder\", \"name\": \"myCompoundDocumentFolder\", \"properties\": {\"dc:title\": \"My title\", \"dc:description\": \" \"}}";
+        try (CloseableClientResponse response = getResponse(RequestType.POST, "path/myCompoundDocument", data)) {
+            assertEquals(SC_CREATED, response.getStatus());
+        }
```

Comment:
Seems useless, already tested in the test before.

Reply:
In the previous test we created a compound doc. Here we created the compound folder type. If this is not added value, I can create the compound folder via the core session like its parent.